tensorboard/plugins/histogram/tf_histogram_dashboard/histogramCore.ts

```diff
@@ -82,20 +82,27 @@ export function backendToIntermediate(
  */
 export function intermediateToD3(
   histogram: IntermediateHistogram,
-  min: number,
-  max: number,
+  min: number | undefined,
+  max: number | undefined,
   numBins = 30
 ): D3HistogramBin[] {
+  if (min === undefined || max == undefined) {
+    min = 0;
+    max = 0;
+  }
   if (max === min) {
-    // Create bins even if all the data has a single value.
+    // If the output range is 0 width, use a default non 0 range for
+    // visualization purpose.
     max = min * 1.1 + 1;
     min = min / 1.1 - 1;
   }
   // Terminology note: _buckets_ are the input to this function,
   // while _bins_ are our output.
   const binWidth = (max - min) / numBins;
   let bucketIndex = 0;
-  return d3.range(min, max, binWidth).map((binLeft) => {
+  let d3HistogramBins: D3HistogramBin[] = [];
```

Comment:
let -> const. You are not changing the reference to the array but rather pushing value into the same array instance.

Reply:
Thanks for this comment! I will apply it in https://github.com/tensorflow/tensorboard/pull/5391.